control/state.py

```diff
@@ -125,9 +195,41 @@ def __init__(self, config):
             self.logger.error(f"Unable to create omap: {ex}. Exiting!")
             raise
 
+    def __exit__(self, exc_type, exc_value, traceback):
+        if self.use_notify:
+            self.watch.close()
+        self.ioctx.close()
+
+    def get_local_version(self) -> str:
+        """Returns local version."""
+        return str(self.version)
+
+    def set_local_version(self, version_update: int):
+        """Sets local version."""
+        self.version = version_update
+
+    def get_omap_version(self) -> str:
+        """Returns OMAP version."""
+        with rados.ReadOpCtx() as read_op:
+            iter, _ = self.ioctx.get_omap_vals_by_keys(read_op,
+                                                       (self.OMAP_VERSION_KEY,))
+            self.ioctx.operate_read_op(read_op, self.omap_name)
+            value_list = list(dict(iter).values())
+            if len(value_list) == 1:
+                val = str(value_list[0], "utf-8")
```

Comment:
Do we want to parse the version here directly as `int` instead since we seem to parse it later as `int` anyway?